cmake/HPX_SetupHwloc.cmake

```diff
@@ -13,9 +13,102 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-find_package(Hwloc)
-if(NOT Hwloc_FOUND)
-  hpx_error(
-    "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+if(NOT HPX_WITH_FETCH_HWLOC)
+  find_package(Hwloc)
+  if(NOT Hwloc_FOUND)
+    hpx_error(
+      "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+    )
+  endif()
+else()
+  set(HPX_WITH_HWLOC_VERSION "2.9")
+  set(HPX_WITH_HWLOC_RELEASE "2.9.3")
+  hpx_info(
+    "HPX_WITH_FETCH_HWLOC=${HPX_WITH_FETCH_HWLOC}, Hwloc v${HPX_WITH_HWLOC_RELEASE} will be fetched using CMake's FetchContent"
   )
+  if(UNIX)
+    include(FetchContent)
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v${HPX_WITH_HWLOC_VERSION}/hwloc-${HPX_WITH_HWLOC_RELEASE}.tar.gz
+      TLS_VERIFY true
+    )
+    if(NOT HWLoc_POPULATED)
+      fetchcontent_populate(HWLoc)
+      execute_process(
+        COMMAND
+          sh -c
+          "cd ${CMAKE_BINARY_DIR}/_deps/hwloc-src && ./configure --prefix=${CMAKE_BINARY_DIR}/_deps/hwloc-installed && make -j && make install"
+      )
+    endif()
+    set(HWLOC_ROOT "${CMAKE_BINARY_DIR}/_deps/hwloc-installed")
+    set(Hwloc_INCLUDE_DIR
+        ${HWLOC_ROOT}/include
+        CACHE INTERNAL ""
+    )
+    if(APPLE)
+      set(Hwloc_LIBRARY
+          ${HWLOC_ROOT}/lib/libhwloc.dylib
+          CACHE INTERNAL ""
+      )
+    else()
+      set(Hwloc_LIBRARY
+          ${HWLOC_ROOT}/lib/libhwloc.so
+          CACHE INTERNAL ""
+      )
+    endif()
+
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows" AND CMAKE_SIZEOF_VOID_P
+                                                       EQUAL 8
+  )
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v${HPX_WITH_HWLOC_VERSION}/hwloc-win64-build-${HPX_WITH_HWLOC_RELEASE}.zip
+      TLS_VERIFY true
+    )
+    if(NOT HWLoc_POPULATED)
+      fetchcontent_populate(HWLoc)
+    endif()
+    set(HWLOC_ROOT
+        "${CMAKE_BINARY_DIR}/_deps/hwloc-src"
+        CACHE INTERNAL ""
+    )
+    include_directories(${HWLOC_ROOT}/include)
+    link_directories(${HWLOC_ROOT}/lib)
+    set(Hwloc_INCLUDE_DIR
+        ${HWLOC_ROOT}/include
+        CACHE INTERNAL ""
+    )
+    set(Hwloc_LIBRARY
+        ${HWLOC_ROOT}/lib/libhwloc.dll.a
+        CACHE INTERNAL ""
+    )
+  else()
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v${HPX_WITH_HWLOC_VERSION}/hwloc-win32-build-${HPX_WITH_HWLOC_RELEASE}.zip
```

Comment:
PTAL https://github.com/SAtacker/hpx-template/blob/ba47fba9bda7833fdbd9698ee5863ddd959e6513/CMakeLists.txt#L108-L119

Reply:
@vrnimje I am pretty sure you need 
```cmake
if(WIN32 OR "${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win64" AND NOT WITH_CUSTOM_HPX)
  add_custom_command(TARGET opt_bench POST_BUILD
    COMMAND ${CMAKE_COMMAND} -E copy_directory
        "${hpx_BINARY_DIR}/${CMAKE_BUILD_TYPE}/bin"
        $<TARGET_FILE_DIR:opt_bench>
  )
  add_custom_command(TARGET opt_bench POST_BUILD
    COMMAND ${CMAKE_COMMAND} -E copy_if_different
        "${HWLOC_ROOT}/bin/libhwloc-15.dll"
        $<TARGET_FILE_DIR:opt_bench>
  )
  ```

Reply:
Added this command in https://github.com/STEllAR-GROUP/hpx/pull/6394/commits/7ac5b3320726b04a109294abfadbb8a406347b60

Reply:
What about 32Bit Windows? Do we support that here as well?

Reply:
Yes, 32-bit Windows is also supported here: 
https://github.com/STEllAR-GROUP/hpx/blob/badead127a2a5721c64fe90a33c4f7b0cefa758c/cmake/HPX_SetupHwloc.cmake#L86-L108
I have also tested it locally using `cmake -A Win32`.